### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by nieopierzony**
Related to #120

The idea was taken from online olympiads.In addition to the known tests, it is also tested on N random ones.  The bad thing is that you have to have 100% working code to create the expected answer. 

Test runner runs tests by task names, which are passed via console arguments. E. g. `npm test -- distinct`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "exams",
+  "version": "1.0.0",
+  "description": "",
```
Comment (tshemsedinov) [inline package.json:4]:
```suggestion
  "private": true,
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "scripts": {
+    "test": "node test/tasks.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nieopierzony/Exams.git"
+  },
+  "keywords": [],
+  "author": "",
```
Comment (tshemsedinov) [inline package.json:13]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "exams",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "test": "node test/tasks.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nieopierzony/Exams.git"
```
Comment (tshemsedinov) [inline package.json:10]:
```suggestion
    "url": "https://github.com/HowProgrammingWorks/Exams"
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  "keywords": [],
+  "author": "",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/nieopierzony/Exams/issues"
```
Comment (tshemsedinov) [inline package.json:16]:
```suggestion
    "url": "https://github.com/HowProgrammingWorks/Exams/issues"
```

File: package.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  "homepage": "https://github.com/nieopierzony/Exams#readme",
+  "devDependencies": {
+    "eslint": "^8.6.0",
+    "prettier": "^2.5.1"
+  }
```
Comment (tshemsedinov) [inline package.json:22]:
```suggestion
  }
```

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
- We do not need random arrays in this tests, just a few predefined examples
- This test should apply all test cases for multiple implementations of each task

File: test/distinct.task.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+const testCases = [
+  { input: [[1, 2, 1, 3, 1, 4]], expected: [1, 2, 3, 4] },
+  { input: [[4, 6, 2, 6, 2]], expected: [4, 6, 2] },
+];
+
+assert.equal(typeof distinct, 'function', ERR_FUNCTION_NOT_IMPORTED);
+
+for (const testCase of testCases) {
+  const cachedInput = JSON.parse(JSON.stringify(testCase.input));
```
Comment (nieopierzony) [inline test/distinct.task.js:17]: For some reason `const cachedInput = [...testCase.input];` doesn't work

**Comment by nieopierzony** [comment]
I decided to use child_process to call the code linter. I could use the npm feature to insert arguments inside the command `npm lint:task --task=distinct -> eslint Tasks%npm_config_task%.js` and it would work, but only on Windows. On Linux is a different templates, so `eslint Tasks/${npm_config_task}.js` but it wouldn't work on Windows ._.
So I decided to find a compromise by using child processes inside the test runner

**Comment by nieopierzony** [comment]
I'm also considering storing cases not in a huge JSON file, but exporting it from each task along with the function

**Comment by tshemsedinov** [comment]
Closed in favour of https://github.com/HowProgrammingWorks/Exams/pull/128 but you can port come test case to this structure

**Pull request closed without merging**